jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -5354,4 +5354,41 @@ public IntStream codePoints() {
 		return value;
 	}
 /*[ENDIF] Java10 */	
+
+/*[IF Java11]*/
+	/**
+	 * Returns a string object containing the character (Unicode code point)
+	 * specified.
+	 * 
+	 * @param codePoint
+	 *            a Unicode code point.
+	 * @return a string containing the character (Unicode code point) supplied.
+	 * @throws IllegalArgumentException
+	 *             if the codePoint is not a valid Unicode code point.
+	 * @since 11
+	 */
+	static String valueOfCodePoint(int codePoint) {
+		String string;
+		if ((codePoint < Character.MIN_CODE_POINT) || (codePoint > Character.MAX_CODE_POINT)) {
+			/*[MSG "K0800", "Invalid Unicode code point"]*/
+			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0800"));	//$NON-NLS-1$
```

Comment:
Would it be possible to put case as the last `else` case as it occurs least often. The JIT will have to do less work when block shuffling and if it every got it wrong we wouldn't pay the performance penalty.

Reply:
Wouldn't the next check for `<= 255` need to expand then?  It would also need to validate codePoint > MIN_CODE_POINT.

Reply:
Will following checking help JIT performance?
```
if ((codePoint < Character.MIN_CODE_POINT)  {
  // error handling 
} else if (codePoint <= 255) {
} else if (codePoint < Character.MIN_SUPPLEMENTARY_CODE_POINT) {
} else if (codePoint <= Character.MAX_CODE_POINT) {
} else {
  // error handling
}
```
This saves the less often check `(codePoint > Character.MAX_CODE_POINT)` to the last.